libclamav_rust/src/fuzzy_hash.rs

```diff
@@ -81,18 +84,18 @@ pub enum FuzzyHash {
 }
 
 impl TryFrom<&str> for ImageFuzzyHash {
-    type Error = &'static str;
+    type Error = Error;
 
     fn try_from(value: &str) -> Result<Self, Self::Error> {
         if value.len() != 16 {
-            return Err("Image fuzzy hash must be 16 characters in length");
+            return Err(Error::InvalidHashLength("ImageFuzzyHash"));
         }
```

Comment:
If you want to avoid repeating yourself...
```suggestion
            return Err(Error::InvalidHashLength(std::any::type_name::<Self>()));
```